bugbug/similarity.py

```diff
@@ -368,6 +369,7 @@ def __init__(
         cleanup_urls=True,
         nltk_tokenizer=False,
         confidence_threshold=0.8,
+        model="word2vec",
```

Comment:
If `Word2VecWmdSimilarity`, `Word2VecWmdRelaxSimilarity` and `Word2VecSoftCosSimilarity` can only work when `model` is doc2vec, I would remove this option and totally separate the `Word2VecSimilarityBase` and `Doc2VecCosSimilarity` classes (maybe create a `Doc2VecCosSimilarity` class and make `Doc2VecCosSimilarity` inherit from it, so we can also introduce other kinds of Doc2Vec classes like for the Word2Vec case).

Otherwise, if we can make it so `Word2VecWmdSimilarity`, `Word2VecWmdRelaxSimilarity` and `Word2VecSoftCosSimilarity` work no matter what the underlying model is (word2vec or doc2vec), then we can have a single base class like you started doing.

Reply:
Yeah, If we are using word embeddings from doc2vec, then we can do that.

But, I just noticed that gensim by default doesn't train word embeddings while using the `PV-DBOW` algorithm. Word embeddings are trained along with doc in the case of `PV-DM` only. Though we can set one argument ( [doc_link](https://radimrehurek.com/gensim/models/doc2vec.html#:~:text=using%20dm_concat%20mode.-,dbow_words,-(%7B1%2C0%7D%2C%20optional)%20%E2%80%93%20If) ) to train word embedding in the case of `PV-DBOW`, but it will make the training very slow.